src/d/actor/d_a_npc_km1.cpp

```diff
@@ -4,248 +4,816 @@
 //
 
 #include "d/actor/d_a_npc_km1.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_s_play.h"
+#include "d/d_snap.h"
 #include "m_Do/m_Do_ext.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
+#include "weak_bss_936_to_1036.h"
+static daNpc_Km1_HIO_c l_HIO;
 
 /* 000000EC-00000144       .text __ct__15daNpc_Km1_HIO_cFv */
 daNpc_Km1_HIO_c::daNpc_Km1_HIO_c() {
-    /* Nonmatching */
-}
 
-/* 00000144-000002F0       .text nodeCallBack_Km__FP7J3DNodei */
-static BOOL nodeCallBack_Km(J3DNode*, int) {
-    /* Nonmatching */
+    static daNpc_Km1_c::prm_tbl a_prm_tbl[] = {0x1FFE38E0, 0xE002C720, 0x00000000, 0x00000000, 
+        0x08000800, 150.0, 0x00000000};
+    memcpy(&mPrmTbl,a_prm_tbl,0x1C);
+    mNo = -1;
+    field_0x8 = -1;
+    return;
 }
 
+/* 00000144-000002F0       .text nodeCallBack_Km__FP7J3DNodei */
+static BOOL nodeCallBack_Km(J3DNode* i_node, int i_calcTiming) {
+
+
+    if (i_calcTiming == 0) {
+        J3DModel* model = j3dSys.getModel();
+        daNpc_Km1_c* km1Actor = (daNpc_Km1_c *)(model->getUserArea());
+        u16 jointIdx;
+        if (km1Actor) {
+            static cXyz a_att_pos_offst(0.0,0.0,0.0);
+            static cXyz a_eye_pos_offst(20.0,20.0,0.0);
+            jointIdx = ((J3DJoint*)(i_node))->getJntNo();
+            mDoMtx_stack_c::copy(model->getAnmMtx(jointIdx));
+            if(jointIdx == km1Actor->getHeadJntNum()){
+                MTXMultVec(mDoMtx_stack_c::get(),&a_att_pos_offst,km1Actor->getAttPos());
+                mDoMtx_stack_c::XrotM(km1Actor->m_jnt.getHead_y());
+                mDoMtx_stack_c::ZrotM(km1Actor->m_jnt.getHead_x()); 
+                MTXMultVec(mDoMtx_stack_c::get(),&a_eye_pos_offst,km1Actor->getEyePos());
+            }else if(jointIdx == km1Actor->getBackboneJntNum()){
+                mDoMtx_stack_c::XrotM(km1Actor->m_jnt.getBackbone_y());
+                mDoMtx_stack_c::ZrotM(km1Actor->m_jnt.getBackbone_x());  
```

Comment:
These are probably what those inlines in the actor class are, wrappers for the inner inlines
```suggestion
                mDoMtx_stack_c::XrotM(km1Actor->getHead_y());
                mDoMtx_stack_c::ZrotM(km1Actor->getHead_x()); 
                MTXMultVec(mDoMtx_stack_c::get(),&a_eye_pos_offst,km1Actor->getEyePos());
            }else if(jointIdx == km1Actor->getBackboneJntNum()){
                mDoMtx_stack_c::XrotM(km1Actor->getBackbone_y());
                mDoMtx_stack_c::ZrotM(km1Actor->getBackbone_x());  
```